src/main/java/org/broadinstitute/hellbender/utils/reference/FastaReferenceWriter.java

```diff
@@ -0,0 +1,665 @@
+     *     The description cannot contain {@link Character#isISOControl(char)}. If set to {@code null} or the empty
+     *     string ("") no description will be outputted.
+     * </p>
+     * <p>
+     *     The input bases-per-line is set to the default provided at construction or {@link #DEFAULT_BASES_PER_LINE}
+     *     if none was provided.
+     * </p>
+     * <p>
+     *     This method cannot be called after the writer has been closed.
+     * </p>
+     * <p>
+     *     It also will fail if no base was added to the previous sequence if any.
+     * </p>
+     * @param sequenceName the name of the new sequence.
+     * @param description optional description for that sequence.
+     * @return this instance.
+     * @throws IllegalArgumentException if any argument does not comply with requirements listed above or if a sequence
+     *  with the same name has already been added to the writer.
+     * @throws IllegalStateException if no base was added to the previous base or the writer is already closed.
+     * @throws IOException if such exception is thrown when writing into the output resources.
+     */
+    public FastaReferenceWriter startSequence(final String sequenceName, final String description)
+        throws IOException
+    {
+        return startSequence(sequenceName, description, defaultBasePerLine);
+    }
+
+    /**
+     * Starts the input of the bases of a new sequence.
+     * <p>
+     *     This operation automatically closes the previous sequence base input if any.
+     * </p>
+     * <p>
+     *     The sequence name cannot contain any blank characters (as determined by {@link Character#isWhitespace(char)}),
+     *     control characters (as determined by {@link Character#isISOControl(char)}) or the the FASTA header star character
+     *     {@value #HEADER_START_CHAR}. It cannot be the empty string either ("").
+     * </p>
+     * <p>
+     *     The description cannot contain {@link Character#isISOControl(char)}. If set to {@code null} or the empty
+     *     string ("") no description will be outputted.
+     * </p>
+     * <p>
+     *     The input bases-per-line must be 1 or greater.
+     * </p>
+     * <p>
+     *     This method cannot be called after the writer has been closed.
+     * </p>
+     * <p>
+     *     It also will fail if no base was added to the previous sequence if any.
+     * </p>
+     * @param sequenceName the name of the new sequence.
+     * @param description optional description for that sequence.
+     * @param basesPerLine number of bases per line for this sequence.
+     * @return this instance.
+     * @throws IllegalArgumentException if any argument does not comply with requirements listed above.
+     * @throws IllegalStateException if no base was added to the previous base or the writer is already closed of
+     *         the sequence has been already added.
+     * @throws IOException if such exception is thrown when writing into the output resources.
+     */
```

Comment:
It seems like we probably also want a method that takes a name and the bases and adds them in a single command.

Reply:
Done.